### ISSUE TITLE
How should I make brute force protection to satisfy this tests?

### ISSUE THREAD

**Issue by dotterian** [logux/backend-test#1]
I'm writing PHP backend for Logux.
My brute force protection prevents trying to log in for 60 seconds after 5 attempts with wrong secret.
Brute force test passes but following test fails due to brute force prevention mechanism.
Any advice on how to make this work?

**Comment by ai** [comment]
What error do you have?

**Comment by dotterian** [comment]
It's not error in backend-test per se. I just cant's seem to get how my backend should block all requests after a number of actions with wrong secret (brute force test) but magically resume responding to requests when this test has passed.

For now my code sets 1 minute timer after each request with wrong secret and keeps counting attempts and after some number of failed attempts starts responding with "Too much requests with wrong secret" for exactly one minute then resumes responding to requests normally.

In code it looks something like this (simplified pseudo-code in JS)
```js
if (wrongSecret) {
  lastFailedAttempt = new Date()
  failedAttemptCounter += 1
}

if (differenceInSeconds(lastFailedAttempt, new Date()) < 60) {
  if (failedAttemptsCounter >= 5) {
    respond("Too many attempts")
  }
} else {
  failedAttemptCounter = 0
}

/* process action */
```

Brute force test passes but next test fails because it runs instantly after previous test and backend still blocks all requests due to brute force prevention.

I've tested this with 1 second timeout but it's still too long and next test fails and timeout is unreasonably short for preventing real brute force.

So it's either my brute force algorithm is wrong (and if so I seek advice on improving it) or we need to add some timeout to tests, thus imposing this exact timeout on backend.

**Comment by ai** [comment]
Got it. I moved the brute-force test to the end and released the fix in 4.0.2 https://github.com/logux/backend-test/commit/0ac089927c3b80f75c935d3e4b68e76a652419f8
